cromwell-backend/src/main/scala/cromwell/backend/provider/local/LocalBackend.scala

```diff
@@ -0,0 +1,348 @@
+      subs => subs.subscriber ! new TaskStatus(Status.Canceled, None))
+  }
+
+  /**
+    * Executes task in given context.
+    */
+  override def execute(): Unit = {
+    subscriptions.filter(subs => subs.eventType.isInstanceOf[ExecutionEvent]).foreach(
+      subs => subs.subscriber ! executeTask)
+  }
+
+  /**
+    * Performs a cleanUp after the task was executed.
+    */
+  override def cleanUp(): Unit = {
+    // Do nothing.
+  }
+
+  /**
+    * Subscribe to events on backend.
+    */
+  override def subscribeToEvent[T](subscription: Subscription[T]): Unit = {
+    val sub = subscription.asInstanceOf[Subscription[ActorRef]]
+    subscriptions += sub
+    sub.subscriber ! Subscribed
+  }
+
+  /**
+    * Unsubscribe to events on backend.
+    */
+  override def unsubscribeToEvent[T](subscription: Subscription[T]): Unit = {
+    val sub = subscription.asInstanceOf[Subscription[ActorRef]]
+    subscriptions -= sub
+    sub.subscriber ! Unsubscribed
+  }
+
+  /**
+    * Get file content hash.
+    * @param files List of files to compute.
+    * @return List of hashes related to files.
+    */
+  override def computeInputFileHash(files: List[Path]): Map[Path, String] = {
+    files.map(path => path -> File(path.toString).md5).toMap[Path, String]
+  }
+
+  /**
+    * Get container image hash.
+    * @param imageName Image name.
+    * @return A hash.
+    */
+  override def computeContainerImageHash(imageName: String): Map[String, String] = ???
+
+  /**
+    * Gather Docker image name from runtime attributes. It it's not present returns none.
+    * @param runtimeAttributes Runtime requirements for the task.
+    */
+  private def getDockerImage(runtimeAttributes: Map[String, String]): Option[String] = {
+    if (runtimeAttributes.filter(p => p._1.equals(DockerFlag)).size == 1 &&
+      !runtimeAttributes.filter(p => p._1.equals(DockerFlag)).values.head.isEmpty) {
+      Some(runtimeAttributes.filter(p => p._1.equals(DockerFlag)).values.head)
+    } else {
+      None
+    }
+  }
+
+  /**
+    * Extracts folder path from specific input file.
+    * @param file Absolute path from input file.
+    * @return File's folder.
+    */
+  private def extractFolder(file: String): Option[String] = {
+    try {
+      Some(file.substring(0, file.lastIndexOf("/")))
+    } catch {
+      case oooe: StringIndexOutOfBoundsException =>
+        logger.warn("Input with no valid folder pattern. It may be a intermediate value.", oooe)
+        None
+      case ex: Exception =>
+        logger.error("Unhandled exception.", ex)
+        throw ex
+    }
+  }
+
+  /**
+    * Creates docker command in order to execute the task into a container.
+    * @param image Docker image name.
+    * @return Command to execute.
+    */
+  private def buildDockerRunCommand(image: String): String = {
+    val dockerVolume = "-v %s:%s"
+    val inputFolder = task.inputs.filter(p => p._2.isInstanceOf[WdlFile]).map(p => extractFolder(p._2.toWdlString.replace("\"", "")))
```

Comment:
Changed to for comprehension. Pros: more descriptive. Cons: add more lines of code.